clients/src/main/java/org/apache/kafka/common/utils/PoisonPill.java

```diff
@@ -24,19 +24,52 @@
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import javax.management.MBeanServer;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Value;
 
 
+class PoisonPillStats {
+    private final Metrics metrics;
+    private final Sensor timeSinceLastDequeueSensor;
+    public PoisonPillStats(Metrics metrics) {
+        this.metrics = metrics;
+
+        String metricGroupName = "PoisonPill-metrics";
+        this.timeSinceLastDequeueSensor = metrics.sensor("time-since-last-dequeue");
+
+        this.timeSinceLastDequeueSensor.add(metrics.metricName("time-since-last-dequeue",
```

Comment:
suggesting adding a unit: `time-since-last-deque-ms` and `time-since-last-deque-ms-max`

Reply:
If no unit is mentioned, the default assumed unit is ms, e.g. https://github.com/linkedin/kafka/blob/071193658b66d1c18a14b2371b226c900032c2a9/clients/src/main/java/org/apache/kafka/clients/consumer/internals/KafkaConsumerMetrics.java#L52

So I think it's ok to not have the "-ms" part.